inclusive nightscout community values

bewest opened the pull request:
Address dev concerns and preferences in https://github.com/nightscout/nightscout.github.io/issues/8.

bewest committed:
make explicit more inclusive values

Hopefully a clarification of developer recommendations.
Remove "Master" from description for the same reasons.

bewest committed:
apply Nightscout values to new users

Provide fuller set of expectations and contrast different ways to get started
with Nightscout.  This provides new users and prospective users of Nightscout a
fuller basis for understanding how to get started and what to expect from
different experiences.

bewest committed:
include T1Pal as a high quality solution

With these patches, people should know that T1Pal reduces the instance of many
of these issues to virtually zero.  This will reduce the tech support burden
for the community.

bewest committed:
give more options for database maintenance

Allow people an escape hatch if they don't want to maintain their database.

bewest committed:
add links for history

Add link for contribution history to cgm-remote-monitor, as
well as youtube video rendering git contributon to
overall Nightscout ecosystem 2011 - 2016.

bewest committed:
typos

psonnera commented:
I thought this documentation was for the DIY project. I understand very well the importance of T1Pal for Nightscout but would prefer a dedicated page explaining all the advantages of the paid version over DIY rather than mentioning it all over the place.

liroyvh commented:
@psonnera Agreed, preferably this documentation should stay free of such advertisements. And if such a comparison is made as you suggest, which could be a good idea, then imho it should not be biased. Moreover, all alternative options should be mentioned; including for example great free managed services such as https://ns.10be.de/en/index.html instead of only T1Pal.

LiroyvH commented:
@psonnera Agreed, preferably the (DIY-)documentation should keep its focus on DIY and be neutral (/ad-free) - not in the least because it can set a precedent and when more (paid) services are introduced: this section of the documentation could grow exponentially (unless NS isn't neutral on the matter.). Imho as it's DIY it also should not be written in a manner that tries to give people gentle nudges to a (specific) non-DIY paid service/demotivate them to use free/DIY options in any way, but the chosen language here feels a bit like it's (perhaps unintentionally) attempting to do so. :) 

And if such a comparison is separately made as you suggest, then imho it should not be biased in any way/shouldn't be geared to one solution. All known alternative options should be mentioned; including for example great free managed services such as https://ns.10be.de/en/index.html instead of only paid options like T1Pal. Such a comparison could be a good idea for sure! :) 

-edit- clarified DIY-portion

sulkaharo commented:
So - IMO the neutral stance is that it's fine for people to get access to Nightscout however they choose, whether that's paid or free. Nightscout has been made to ease the burden of diabetes management and the more people have access to Nightscout and the more organisations have vested interest in keeping it running, the better.  How any paid service profits from the hosting or contributes to the development is different discussion.

I agree the sections that has the notes on DIY setup vs T1Pal should be rewritten to some form that makes the differentiation based on DIY hosting and a paid service, without being explicit to T1Pal. Having a list of paid services on the section of how to get Nightscout is IMO completely fine. If 10be admins want their site to be listed, they can always PR in a mention - I've assumed they don't want to be here due to not asking for a link.

marionbarker commented:
I concur with both comments above and I have a suggestion.

NS is a wonderful tool I use every day (since 2015).  I am also a user of T1Pal - I'm glad to know it exists - glad to have someone else maintain it for me so I can focus on other things.

I suggest that for the other pages where Ben suggested adding an admonition, I would propose instead:
```markdown
!!! note "DIY Only"
    These instructions apply to DIY setup only.  
    For [Paid Services](../index.md#paid-solutions), check with the provider of that service.
```

bewest commented:
Good suggestions, thanks.  The main thrust here is to edit the document to apply more inclusive values.  The notes and admonitions have been designed to facilitate common usage.  When someone posts on FB that "my NS got messed up," the solution has been to toss them a link to either a.) new_user, b.) rebuild stack, c.) reconnect mongo, d.) trouble shoot connection string, regardless of their interest in solving that technical issue, and often without any additional context.  There have been complaints that when people mention trouble with T1Pal, they don't know what to do, these edits additionally make clear when and how to contact T1Pal using the same specificity and logic as the Heroku details.  I feel these edits address both these issues in reasonably informing people of their options given the circumstances.  I find the paid vs DIY distinctions confounding because Heroku and MongoDB barrage Nightscout users with sales offers with profitable results or they wouldn't do it.  There is no separating the Heroku instructions from profit for Salesforce.  I would appreciate help assembling a matrix of solutions, the steps and price to get started vs sustain operations, as well as what to expect from each one.  For example, Digital Ocean approached us to talk about their $5 PaaS product, App Platform.  I used the values in this document to formulate a response talking about the needs across our community in simplicity, predictability, and debugging issues responsibility.  Terry has expressed interest in adding a paid Azure section to support Medtronic Carelink.  However, these patches are trying to stay small in scope as possible.  There will be more discussions to come.

bewest committed:
add typographical emphasis

psonnera commented:
@liroyvh ns10be is the most beautiful example of an open heart easy to access Nightscout solution to me. Unfortunately it's not sized for the whole community and advertising for it here will lead to its collapse imho. I've pointed this discussion to @zehnBE to have his feedback.

bewest commented:
Maybe I am getting hung up on the term "paid services" in a general way while it implies DIY is something different.  People tell us they pay for DO, Azure, AWS, Heroku, and MongoDB, Pushover and these are all DIY.  I have tried to continue the practice of letting each page be a standalone solution.  People don't like clicking through to get a sense of options or alternates.  In addition, I've tried to remain neutral in explaining the benefits for why someone might want either choice as succinctly as possible by mirroring the pre-existing language.  Once we move beyond "choice A and B" and towards a holistic spectrum of "choices A, B, C, D, E ...", I agree more restructuring makes sense.  Are there a few rewording suggestions or deletions where we can avoid restructuring or adding pages in the meantime?

bewest committed:
distinguish between DIY versus nightscout as a service

Nightscout as a service better communicates the departure from DIY.  There is
such a thing as paid DIY which is the default for many spoken, silent, or
otherwise.  While I find distasteful the specific zeal for Heroku, which offers
no support to developers or community to support our concerns, the suggestion
has been to remove direct references to T1Pal.

bewest commented:
@marionbarker thanks for suggestion.  I got over most of my issues by renaming "Paid support" to "Nightscout as a service."  If cost is one dimension managed vs DIY is a second, independent dimension.  With this category, it better prepares where to include ns10be.  If Martin would care for a link, it would go under Nightscout as a service.

StephenBrown2 commented on docs/index.md:
```diff
@@ -9,35 +9,77 @@
+
+There are several parts to this system. You need somewhere online to
+store/process/visualise this data (**a Nightscout Site**), something to upload
+CGM data to your Nightscout (**a Uploader**), and then optionally you can use
```
a/an ("an" preceding a vowel, "a" otherwise)
```suggestion
CGM data to your Nightscout (**an Uploader**), and then optionally you can use
```

StephenBrown2 commented on docs/index.md:
```diff
@@ -9,35 +9,77 @@
+#### Uploader
 
-The CGM data is usually sent to Nightscout by an **uploader device** (often known as a **"Master"**) such as a mobile/cellphone running an app such as xDrip+, Spike, xDrip4iOS or Medtronic Uploader. Alternatively, the CGM data can be pulled directly from your online Dexcom or Medtronic CareLink accounts. The application you'll need for uploading the data depends on which CGM device you're using a ...
+The CGM data is usually sent to Nightscout by an **uploader device** (often known as a **"Uploader"**) such as a mobile/cellphone running an app such as xDrip+, Spike, xDrip4iOS or Medtronic Uploader. Alternatively, the CGM data can be pulled directly from your online Dexcom or Medtronic CareLink accounts. The application you'll need for uploading the data depends on which CGM device you're using ...
```
a/an
```suggestion
The CGM data is usually sent to Nightscout by an **uploader device** (often known as an **"Uploader"**) such as a mobile/cellphone running an app such as xDrip+, Spike, xDrip4iOS or Medtronic Uploader. Alternatively, the CGM data can be pulled directly from your online Dexcom or Medtronic CareLink accounts. The application you'll need for uploading the data depends on which CGM device you're using and what type of mobile device you have. Once you've read this page (yes - all the way to the bottom!), you can find more information here about [Uploaders](./uploader/uploaders.md).
```

bewest committed:
s/a Uploader/an Uploader/g

Thanks to @StephenBrown2, fix grammatical a vs an.
